python/ray/train/tests/test_train_head.py

```diff
@@ -0,0 +1,55 @@
+import os
+import sys
+import time
+
+import pytest
+import requests
+
+import ray
+from ray.train import RunConfig, ScalingConfig
+from ray.train.torch import TorchTrainer
+
+
+@pytest.mark.skipif(
+    os.environ.get("RAY_MINIMAL") == "1",
+    reason="This test is not supposed to work for minimal installation.",
+)
+def test_get_train_runs(monkeypatch, shutdown_only):
+    monkeypatch.setenv("RAY_TRAIN_ENABLE_STATE_TRACKING", "1")
+    try:
+        ray.init(num_cpus=8)
+
+        def train_func():
+            print("Training Starts")
+            time.sleep(0.5)
+
+        datasets = {"train": ray.data.range(100), "val": ray.data.range(100)}
+
+        trainer = TorchTrainer(
+            train_func,
+            run_config=RunConfig(
+                name="my_train_run", storage_path="/tmp/cluster_storage"
+            ),
+            scaling_config=ScalingConfig(num_workers=4, use_gpu=False),
+            datasets=datasets,
+        )
+        trainer.fit()
+
+        # Call the train run api
+        url = ray._private.worker.get_dashboard_url()
+        resp = requests.get("http://" + url + "/api/train/runs")
+        assert resp.status_code == 200
+        body = resp.json()
+        assert len(body["train_runs"]) == 1
+        assert body["train_runs"][0]["name"] == "my_train_run"
+        assert len(body["train_runs"][0]["workers"]) == 4
+
+    finally:
+        del os.environ["RAY_TRAIN_ENABLE_STATE_TRACKING"]
```

Comment:
nit: can remove it as monkeypatch will automatically unset the environment variable after exiting.